obspy/sac/tests/test_core.py

```diff
@@ -478,6 +485,262 @@ def test_read_with_fsize(self):
         self.assertEqual(tr.stats.sac.kstnm, 'CDV     ')
         self.assertEqual(tr.stats.sac.kcmpnm, 'Q       ')
 
+    def test_read_sac_from_bytes_io(self):
+        """
+        Tests reading from a BytesIO object.
+        """
+        with io.BytesIO() as buf:
+            # Read file to BytesIO.
+            with open(self.file, "rb") as fh:
+                buf.write(fh.read())
+            buf.seek(0, 0)
+
+            # Attempt to read from it.
+            tr = readSAC(buf)[0]
+
+        # Open file normally and make sure the results are identical.
+        tr2 = readSAC(self.file)[0]
+        np.testing.assert_array_equal(tr.data, tr2.data)
+        self.assertEqual(tr, tr2)
+
+    def test_read_sac_from_open_file(self):
+        """
+        Tests reading from an open file.
+        """
+        with open(self.file, "rb") as fh:
+            # Attempt to read from it.
+            tr = readSAC(fh)[0]
+
+        # Open file normally and make sure the results are identical.
+        tr2 = readSAC(self.file)[0]
+        np.testing.assert_array_equal(tr.data, tr2.data)
+        self.assertEqual(tr, tr2)
+
+    def test_read_write_bytes_io(self):
+        """
+        Tests reading and writing to and from BytesIO.
+        """
+        st = readSAC(self.file)
+        with io.BytesIO() as buf:
+            writeSAC(st, buf)
+            buf.seek(0, 0)
+            # Attempt to read from it.
+            st2 = readSAC(buf)
+
+        tr = st[0]
+        tr2 = st2[0]
+        # depmen is different as it is actually calculated on the fly.
+        del tr.stats.sac.depmen
+        del tr2.stats.sac.depmen
+        np.testing.assert_array_equal(tr.data, tr2.data)
+        self.assertEqual(tr, tr2)
+
+    def test_read_write_open_file(self):
+        st = readSAC(self.file)
+
+        with NamedTemporaryFile() as tf_out:
+            with open(tf_out.name, "wb") as fh:
```

Comment:
Is it really necessary to re-open the file? `tf_out` should already have the file opened read/write.